### PR TITLE
fix(image): can't change image after set status to 'error'

### PR DESCRIPTION
**Pull request by huangkairan**
fix: https://github.com/ant-design/ant-design/issues/29112

### CODE REVIEW

**Comment by shaodahong** [comment]
Please add test case

**Comment by huangkairan** [comment]
> Please add test case
why when I commit, see this ? I never change change the version
image

**Comment by shaodahong** [comment]
This lint fails cause by eslint low version, you can use no-verify to commit

**Comment by huangkairan** [comment]
> This lint fails cause by eslint low version, you can use no-verify to commit

how to do no-verify commit hahhah ?

**Comment by shaodahong** [comment]
`git commit --no-verify`

**Comment by huangkairan** [comment]
> `git commit --no-verify`

sorry, i dont know how to test my code, i never learn jest ..

**Comment by shaodahong** [comment]
Please resolve conflicting

**Comment by afc163** [comment]
ci broken

File: tests/fallback.test.tsx [hunk 1]
```diff
@@ -59,6 +59,7 @@ describe('Fallback', () => {
     expect(wrapper.find('.rc-image-img').prop('src')).toBe(
       'https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg',
     );
+  });
```
Comment (afc163) [inline tests/fallback.test.tsx:62]:
```suggestion
  });

```

**Comment by huangkairan** [comment]
cant test after resolved conflicts....

File: src/Image.tsx [hunk 2]
```diff
@@ -179,6 +179,9 @@ const ImageInternal: CompoundedComponent<ImageProps> = ({
     if (isCustomPlaceholder) {
       setStatus('loading');
     }
+    if (isError) {
+      setStatus('normal');
```
Comment (shaodahong) [inline src/Image.tsx:183]: 有点不对这个，会把上面的状态冲掉，这个优先级比上面的低，应该放在上面

Comment (huangkairan) [reply src/Image.tsx:183]: 好的，我修改一下

File: src/Image.tsx [hunk 3]
```diff
@@ -176,6 +176,9 @@ const ImageInternal: CompoundedComponent<ImageProps> = ({
   // Keep order end
 
   React.useEffect(() => {
+    if (isError) {
+      setStatus('normal');
+    }
     if (isCustomPlaceholder) {
       setStatus('loading');
     }
```
Comment (afc163) [inline src/Image.tsx:184]: useEffect 依赖里是不是应该加上 isError 和 isCustomPlaceholder

Comment (shaodahong) [reply src/Image.tsx:184]: 这里的逻辑是 src 变化的时候重置状态，如果加上 `isError` `isCustomPlaceholder` 的依赖变化就不对了

Comment (huangkairan) [reply src/Image.tsx:184]: 这里只需要依赖imgSrc吧

**Comment by huangkairan** [comment]
这个bug还准备跟吗。。已经有冲突了

**Comment by huangkairan** [comment]
。。。代码拉乱了 我重新开一个PR吧

**Comment by huangkairan** [comment]
更新了一下和主仓库的代码同步

**Review by shaodahong — APPROVED** [review]
LGTM

**Comment by shaodahong** [comment]
+ rc-image@5.2.2

**Pull request merged**